feat: add wemos-s2-mini

mlg556 opened the pull request:
added the [Wemos S2 Mini](https://www.wemos.cc/en/latest/s2/s2_mini.html) board, currently working with micropython. it seems to work without any issues, [here is a simple wokwi example](https://wokwi.com/projects/354845328269990913) using the SSD1306 I2C display and blinking the onboard LED. for the default i2c and spi pins, I also added their names, for example `35(SCL)` although these labels are not present on the board silkscreen, they are only present in the schematic files.

I would also like to know how to make it work with CircuitPython, [since the board is supported there too](https://circuitpython.org/board/lolin_s2_mini/), and let me know if there are any changes you think should be made.

mlg556 committed:
feat: add wemos-d2-mini

urish commented:
Thanks! Can you please make the SVG file smaller? We aim to keep all boards under 100kb, to ensure that diagrams load fast and do not slow down the browser.

One way to optimize it is using this website: https://jakearchibald.github.io/svgomg/

Another that usually works well is removing small details (such as tiny pins) that have complex shapes.

urish commented:
p.s. I see circuit python calls this board "LOLIN S2 Mini" (in the link you shared), which is different from "Wemos S2 Mini". What's the actual name of the board?

mlg556 commented:
> Can you please make the SVG file smaller? We aim to keep all boards under 100kb, to ensure that diagrams load fast and do not slow down the browser.

I didn't think of that, makes sense! let me get rid of a few details.

> I see circuit python calls this board "LOLIN S2 Mini" (in the link you shared), which is different from "Wemos S2 Mini". What's the actual name of the board?

it's confusing, it looks like their first board with ESP8266 was called [Lolin D1 Mini](https://www.wemos.cc/en/latest/d1/d1_mini.html), and the name kinda stuck for other models too? although [there is no mention](https://www.wemos.cc/en/latest/s2/s2_mini.html) of "Lolin" in the S2 board/its documentation. apparently PlatformIO calls it [WEMOS LOLIN S2 Mini](https://docs.platformio.org/en/latest/boards/espressif32/lolin_s2_mini.html), and they use "lolin_s2_mini" in the configs. which one should we pick?

urish commented:
As usual, naming things is hard :) 

If the manufacturer calls it S2 Mini, then I think `wemos-s2-mini` is a sensible name.

mlg556 committed:
fix(wemos-s2-mini): reduced svg size

mlg556 commented:
I've reduced the svg size to 70kb and tested it again, seems to be working.

>  then I think wemos-s2-mini is a sensible name.

I think so too, but it's interesting why everyone else but the manufacturer is insisting on "Lolin" :D

urish commented:
Thanks for updating the PR!

Few more notes:
1. The fqbn should be esp32:esp32:lolin_s2_mini
2. I looked at [the schematic](https://www.wemos.cc/en/latest/_static/files/sch_s2_mini_v1.0.0.pdf), and TX/RX pins do not exist on this board. Rather, it uses Serial over USB to communicate.

mlg556 committed:
fix(wemos-s2-mini): fixed fqbn, removed rx-tx pins

mlg556 commented:
fixed the fqbn and removed TX/RX pins.

> I looked at [the schematic](https://www.wemos.cc/en/latest/_static/files/sch_s2_mini_v1.0.0.pdf), and TX/RX pins do not exist on this board. Rather, it uses Serial over USB to communicate.

yes I was confused about that part, I thought this board doesn't expose RX/TX pins as pads but they are connected to the USB-C connector, but it looks like they are not connected at all.

urish commented:
Awesome! thanks for the quick turnaround.

Here's a quick demo project I made to test the new board:

https://wokwi.com/projects/355047217294313473

mlg556 commented:
amazing! thanks for the useful feedback and corrections. I might add some more boards (like seeed xiao or tinypico) in the future if I have more free time, we'll see. wokwi is such a cool project!

urish commented:
Thank you! Also added the new board to the [official documentation](https://docs.wokwi.com/guides/esp32#esp32-boards).

I'd love to see more boards from you in the future :-)
